algosdk/v2client/algod.py

```diff
@@ -188,6 +188,34 @@ def send_raw_transaction(self, txn, **kwargs):
         headers = { 'Content-Type': 'application/x-binary' }
         return self.algod_request("POST", req, data=txn, headers=headers, **kwargs)["txId"]
 
+    def wait_for_confirmation(self, transaction_id, timeout):
+        """
+        Wait until the transaction is confirmed or rejected, or until 'timeout'
+        number of rounds have passed. 
+
+        Args:
+            transaction_id (str): the transaction to wait for
+            timeout (int): maximum number of rounds to wait
+        
+        Returns:
+            dict: pending transaction information, or None if the transaction
+                is not confirmed or rejected in the next timeout rounds
+
+        Raises:
+            AlgodHTTPError: if the transaction is not found
+        """
+        start_round = self.status()["next-version-round"]
+        current_round = start_round
+
+        while current_round < start_round + timeout:
+            self.status_after_block(current_round)
+            pending_txn = self.pending_transaction_info(transaction_id)
```

Comment:
Some customers may be running algod behind a load balancer. In that situation, `pending_transaction_info` might throw an exception because the transaction_id isn't in the transaction pool but we should continue waiting until the timeout.